pkg/flow/internal/controller/node_declare.go

```diff
@@ -0,0 +1,51 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/grafana/river/ast"
+	"github.com/grafana/river/vm"
+)
+
+type DeclareNode struct {
+	label         string
+	nodeID        string
+	componentName string
+	declare       *Declare
+	mut           sync.RWMutex
+}
+
+var _ BlockNode = (*DeclareNode)(nil)
+
+// NewDeclareNode creates a new declare node with a content which will be loaded by declare component nodes.
+func NewDeclareNode(declare *Declare) *DeclareNode {
+	return &DeclareNode{
+		label:         declare.Block.Label,
+		nodeID:        BlockComponentID(declare.Block).String(),
+		componentName: declare.Block.GetBlockName(),
+		declare:       declare,
+	}
+}
+
+func (cn *DeclareNode) Declare() *Declare {
+	cn.mut.Lock()
+	defer cn.mut.Unlock()
+	return cn.declare
```

Comment:
Why do we need a mutex for Declare? 
If we do need it, why is it not inside the Declare that should encapsulate and guard the access to its own fields?

Reply:
I spent a lot of time trying to see if the fields inside `*Declare` can change and thus need to be protected with a mutex. So far it seems to me that all the `Declare` structs are created inside `flow.ParseSource` and are not modified on updates. We must make it easier to read this code with better naming (or comments if we fail at naming) and remove any misleading / dead code.

Reply:
thanks for bringing my attention to this. This branch has been through a lot of changes with the different design and renaming and feedbacks, and while I'm trying to clean it up there might be artifacts of previous implementations or copy pasta that I need to clean. I will take some time tomorrow to go deeper into details (fields that can be private, fields that don't need mutex, areas where comments are missing etc...)